react/getting_started_with_react/what_is_jsx/what_is_jsx.md

```diff
@@ -0,0 +1,233 @@
+
+1. Return a single root element.
+
+   If you wish to return multiple elements in a component, you can do so by wrapping them in a parent tag. This can be a `<div>`, or, if you don't want the elements to have a container, you could use a [React fragment](https://beta.reactjs.org/apis/react/Fragment), like so: `<>Children</>`
+
+   Correct:
+
+   ~~~jsx
+   function App() {
+     return (
+       <>
+         <h1>Example h1</h1>
+         <h2>Example h2</h2>
+       </>
+       // Could replace <></>with <div></div>
+     );
+   }
+   ~~~
+
+   Incorrect:
+
+   ~~~jsx
+   function App() {
+     return (
+         <h1>Example h1</h1>
+         <h2>Example h2</h2>
+     );
+   }
+   ~~~
+
+2. Close all tags.
+
+   In HTML, many tags are self-closing and self-wrapping. In JSX however, we must explicitly close and wrap these tags.
+
+   `<input>` would become `<input />`, and `<li>` would become `<li></li>`
+
+   Correct:
+
+   ~~~jsx
+   function App() {
+     return (
+       <>
+         <input />
+         <li></li>
+       </>
+     );
+   }
+   ~~~
+
+   Incorrect:
+
+   ~~~jsx
+   function App() {
+     return (
+         <>
+          <input>
+          <li>
+         </>
+     );
+   }
+   ~~~
+
+3. camelCase **Most** things.
+
+   JSX turns into JavaScript, and attributes of elements become keys of JavaScript objects, so you can't use dashes or reserved words such as `class`. Because of this, many HTML attributes are written in camelCase. Instead of `stroke-width`, you'd use `strokeWidth`, and instead of `class` you'd use `className`.
+
+   Correct:
+
+   ~~~jsx
+   function App() {
+     return (
+       <>
+         <div className="container">
+           <svg>
+             <circle cx="25" cy="75" r="20" stroke="green" strokeWidth="2" />
+           </svg>
+         </div>
+       </>
+     );
+   }
+   ~~~
+
+   Incorrect:
+
+   ~~~jsx
+   function App() {
+     return (
+       <>
+         <div class="container">
+           <svg>
+             <circle cx="25" cy="75" r="20" stroke="green" stroke-width="2" />
+           </svg>
+         </div>
+       </>
+     );
+   }
+   ~~~
+
+### Converting HTML to JSX
+
+Now that we've covered the Rules of JSX, we'll go through the conversion of a chunk of HTML to JSX.
+
+~~~jsx
+<h1>Test title</h1>
+<ol class="test-list">
+  <li>List item 1
+  <li>List item 2
+  <li>List item 3
+</ol>
+<svg >
+   <circle cx="25" cy="75" r="20" stroke="green" stroke-width="2" />
+</svg>
+<form><input type="text"></form>
+~~~
+
+If you try to return this from a React component, you would get many errors, so we are going to fix that! We would recommend following along with this in your own React
+environment.(If you don't want to create a new React app locally, go to [react.new](https://react.new/) to have a quick React environment up and going in your web browser.
```

Comment:
Learners should have a React project setup locally from the previous lessons. Although I do like the exposing them to react.new

Wording suggestion to accommodate the above:
```suggestion
If you try to return this from a React component, you would get many errors, so we are going to fix that! We would recommend following along with this in your own React environment.

 Make sure to follow along this example in your local environment. Alternatively, you can go to [react.new](https://react.new/) to have a quick React environment up and going in your web browser.
```

Reply:
Seems like a sound change, will commit the suggestion 👍🏻